compiler/runtime.go

```diff
@@ -38,7 +38,11 @@ func (c compiledFn) Type() semantic.MonoType {
 func (c compiledFn) Eval(ctx context.Context, input values.Object) (values.Value, error) {
 	inputScope := nestScope(c.parentScope)
 	input.Range(func(k string, v values.Value) {
+		// Retaining `v` here because we're storing it in the scope,
```

Comment:
This is a good point. I'm not sure this is needed because accessing the scope should retain the value already. If there's no accompanying release associated with this retain, it likely means we should not use retain here.